includes/class-wc-google-analytics-js.php

```diff
@@ -455,6 +455,10 @@ function remove_from_cart() {
 	 * Tracks a product detail view
 	 */
 	function product_detail( $product ) {
+		if ( ! empty( $product ) ) {
```

Comment:
shouldn't this check if product is empty? because the following lines use the product instance.